components/Search.js

```diff
@@ -24,23 +24,14 @@ class Search extends Component {
 
   state = {
     format: 'list',
-    value: '',
-    hasContent: false,
   };
 
   toggleClearIcon = evt => {
-    const hasContent = evt.target.value.length > 0;
-    this.setState({
-      hasContent,
-      value: evt.target.value,
-    });
+    this.input.value = evt.target.value;
```

Comment:
Once this PR is merged you can trigger things when the value of the `input` using the `onChange` prop which will be passed to `input`.

```jsx
<Search onChange={e => console.log(e.value) } />
```